### ISSUE TITLE
/setuid endpoint does not set 'uids' cookie when both GDPR and GPP is set in the request

### ISSUE THREAD

**Issue by ashishshinde-pubm** [prebid/prebid-server#3166]
Following endpoint returns incorrect response and response-header when both gdpr and gpp parameters are passed in the request.
**Request** 
```
curl -I --location --request GET 'http://localhost:8001/setuid?bidder=pubmatic&uid=123&gpp_sid=1,2&gpp=DBABMA~CPXxRfAPXxRfAAfKABENB-CgAAAAAAAAAAYgAAAAAAAA&gdpr=1&gdpr_consent=BONciguONcjGKADACHENAOLS1rAHDAFAAEAASABQAMwAeACEAFw'

```
**Response-Body**
```
Warning: 'gpp' value will be used over the one found in the deprecated 'gdpr_consent' field.
```
**Response-Header**
```
HTTP/1.1 200 OK
Date: --------
Content-Length: 92
Content-Type: text/plain; charset=utf-8
```
As per the test case [description](https://github.com/prebid/prebid-server/blob/master/endpoints/setuid_test.go#L218) , if both GPP and GDPR are passed in the request then we should give priority to GPP and throw warning.
However, the actual [code](https://github.com/prebid/prebid-server/blob/master/endpoints/setuid.go#L128C4-L128C4) calls the 'Write' method on response-writer first and then writes warning on same response-writer ([here](https://github.com/prebid/prebid-server/blob/master/endpoints/setuid.go#L185)) if responseFormat "f" is set to "i"

Also, here we are trying to call WriteHeader function after write method [code](https://github.com/prebid/prebid-server/blob/master/endpoints/setuid.go#L128C4-L128C4) which is not correct/recommended. 

**Possible Fix**

1. I think we should not throw the warning in response body. If it is required then we can throw warning in response header instead of response body.
2. Do not allow to proceed if both GDPR and GPP are set and return from [here](https://github.com/prebid/prebid-server/blob/master/endpoints/setuid.go#L128)

@guscarreon  Can you please check if this is valid bug.

**Comment by bretg** [comment]
PR https://github.com/prebid/prebid-server/pull/3165

**Comment by SyntaxNode** [comment]
Implemented in PBS-Go 2.2
